docs/enhancements/api-for-skipping-validations.md

```diff
@@ -0,0 +1,153 @@
+---
+title: api-for-skipping-validations
+authors:
+  - "@pmaidmen"
+creation-date: 2023-01-08
+last-updated: 2023-01-08
+---
+
+# Allow validations to be skipped
+
+# Summary
+
+We need a new api that will allow us to skip cluster/host validations, this will be for unsupported use.
+This API should work on a per cluster basis and will have it's own feature flag to allow us to enable it for specific users.
+An event should be emitted whenever the API is used during an installation, this can be used to drive signatures etc.
+# Motivation
+
+Some customer and partners have very specific HW that doesn't pass our validations and we want to allow them to proceed with installation.
+Sometimes we have bugs in our validations that block people from installing and we don't want our partners left in a position where they can't install.
+
+### Goals
+
+- Allow validations to be bypassed so that installation may proceed.
+
+### Non-Goals
+
+- None identified at this stage
+
+## Proposal
+
+I think the API should be kept reasonably simple.
+
+### List existing validations.
+
+GET api/v2/cluster/<id>/skipped-validations
+Will return a JSON array of all the validations that are currently being skipped, format similar to
+
+```
+{
+  "host-skipped-validations": [
+    "no-ip-collisions-in-network",
+    "sufficient-network-latency-requirement-for-role'"
+  ],
+  "cluster-skipped-validations": [
+    "network-prefix-valid",
+    "api-vips-defined"
+  ]
+}
+```
+
+### Add some skipped validations.
+
+POST api/v2/cluster/<id>/skipped-validations
+
+Essentially a document will be submitted via POST in JSON
+```
+{
+  "skipped-host-validations": [
+    "no-ip-collisions-in-network"
+  ],
+  "skipped-cluster-validations": [
+    "api-vips-defined"
+  ]
+}
+```
+
+* Any entries found will be added to the existing list of validations to be skipped. This will appropriately de-deuplicated and no error will be returned if duplication is attempted.
```

Comment:
It would be simpler if we can avoid a custom merging strategy in our API that requires reading documentation to understand.

What about using standard REST more deeply?

```
PUT api/v2/cluster/<id>/skipped-host-validations/no-ip-collisions-in-network

GET api/v2/cluster/<id>/skipped-host-validations/

[
    "no-ip-collisions-in-network"
]

DELETE api/v2/cluster/<id>/skipped-host-validations/no-ip-collisions-in-network
```

Reply:
I like this, a good suggestion I think